### PR TITLE
Remove Linux distro-specific dependencies

### PR DESCRIPTION
**Pull request by eerhardt**
Removing the Linux distros from the runtime.json dependency list. This will cause anyone using self-contained applications to use the portable linux assets by default.

@weshaggard @gkhanna79 @ericstj 

/cc @Petermarcu

### CODE REVIEW

File: pkg/projects/dir.props [hunk 1]
```diff
@@ -108,6 +108,20 @@
     </BuildRID>
   </ItemGroup>
 
+  <!-- RIDs to remove from the runtime dependencies, so the lineup packages don't have distro specific dependencies.
+       Instead, the linux-x64 runtime will be used on all linux distros by default. 
+       NOTE: There isn't a linux-arm build yet, so we can't remove any arm RIDs. -->
```
Comment (gkhanna79) [inline pkg/projects/dir.props]: Can you please file an issue for this? I have brought up linux-arm CoreCLR yesterday (https://dotnet.myget.org/feed/dotnet-core/package/nuget/runtime.linux-arm.Microsoft.NETCore.Runtime.CoreCLR) and we should track this work.

File: pkg/projects/dir.props [hunk 2]
```diff
@@ -108,6 +108,20 @@
...
+  <ItemGroup>
```
Comment (weshaggard) [inline pkg/projects/dir.props]: Should we make this inclusive instead of exclusive? That way we don't have anything creep in that we don't want.

**Comment by weshaggard** [comment]
Do we have a follow-up issue to track the clean up of this along with the removal of the specific RIDs once we verify the portable linux stuff works everywhere we want?

File: pkg/projects/dir.targets [hunk 3]
```diff
@@ -89,4 +89,20 @@
               ItemName="Dependency" />
     </MSBuild>
   </Target>
+
+  <!-- Remove the linux distro specific dependencies so the portable linux RID is used by default. -->
+  <Target Name="RemoveLinuxDistroDependencies"
+          Condition="'@(RuntimeDependency)' != ''"
+          AfterTargets="DetermineRuntimeDependencies"
+          Outputs="%(LinuxDistroRIDNonDependency.Identity)">
```
Comment (weshaggard) [inline pkg/projects/dir.targets]: What do we need to do batching here? I would expect we could do this all in the item group below.

File: pkg/projects/dir.targets [hunk 4]
```diff
@@ -89,4 +89,20 @@
...
+    <PropertyGroup>
+      <CurrentLinuxDistroRID>%(LinuxDistroRIDNonDependency.Identity)</CurrentLinuxDistroRID>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_RuntimeDependencyToRemove Include="@(RuntimeDependency)"
+                                  Condition="'%(RuntimeDependency.TargetRuntime)' == '$(CurrentLinuxDistroRID)'" />
```
Comment (weshaggard) [inline pkg/projects/dir.targets]: Something along the lines of this should allow you to remove the batching from target.
`Condition="'%(RuntimeDependency.TargetRuntime)' == '%(CurrentLinuxDistroRID.Identity)'"`

File: pkg/projects/dir.props (hunk 2 above)
Comment (eerhardt) [inline pkg/projects/dir.props]: Changed to inclusive.

File: pkg/projects/dir.targets (hunk 3 above)
Comment (eerhardt) [inline pkg/projects/dir.targets]: With the change to inclusive, I was able to get rid of the batching.

File: pkg/projects/dir.props (hunk 1 above)
Comment (eerhardt) [inline pkg/projects/dir.props]: https://github.com/dotnet/core-setup/issues/1832

**Comment by eerhardt** [comment]
Closing in favor of https://github.com/dotnet/core-setup/pull/1837

**Pull request closed without merging**